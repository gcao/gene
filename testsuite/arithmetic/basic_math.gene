#!/usr/bin/env gene run
# TEST: Basic arithmetic operations

# Addition
(println "2 + 3 = " (+ 2 3))
(println "10 + -5 = " (+ 10 -5))

# Subtraction  
(println "7 - 4 = " (- 7 4))
(println "3 - 8 = " (- 3 8))

# Multiplication
(println "6 * 7 = " (* 6 7))
(println "3 * -4 = " (* 3 -4))

# Division
(println "15 / 3 = " (/ 15 3))
(println "7 / 2 = " (/ 7 2))  # Integer division

# Modulo - TODO: Not implemented yet
# (println "17 % 5 = " (% 17 5))
# (println "20 % 4 = " (% 20 4))