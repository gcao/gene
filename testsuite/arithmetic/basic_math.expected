2 + 3 =  5
10 + -5 =  5
7 - 4 =  3
3 - 8 =  -5
6 * 7 =  42
3 * -4 =  -12
15 / 3 =  5.0
7 / 2 =  3.5